File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] leftArray = new int [n];
        int[] rightArray = new int[n];

        leftArray[0] = 1;
        rightArray[n-1] = 1;

        for (int i =1; i < n; i++){
            leftArray[i] = nums[i-1] * leftArray[i-1];
        }

        for(int i = n -2; i>=0; i--){
              rightArray[i] = nums[i+1] * rightArray[i+1];
        }
        
        for (int i =0; i<n; i++){
            nums[i] = leftArray[i] * rightArray[i];
        }
        return nums;
    }
}  

File: Data Structures & Algorithms/maximum-sum-circular-subarray/submission-0.py
class Solution:
    def maxSubarraySumCircular(self, nums: List[int]) -> int:
        currMax=0
        currMin=0
        best =nums[0]
        lowest = nums[0]
        total =0

        for num in nums:
            currMax = max(currMax+num,num)
            currMin = min(currMin+num, num)
            total+=num
            best = max(currMax,best)
            lowest = min(currMin,lowest)
        
        return max(best,total-lowest) if best > 0 else best


